lib/client/api.go

```diff
@@ -1885,14 +1889,24 @@ func (tc *TeleportClient) SSH(ctx context.Context, command []string, runLocally
 		return trace.BadParameter("no target host specified")
 	}
 
-	nodeClient, err := proxyClient.ConnectToNode(
+	nodeClient, connectErr := proxyClient.ConnectToNode(
 		ctx,
 		NodeAddr{Addr: nodeAddrs[0], Namespace: tc.Namespace, Cluster: siteInfo.Name},
 		tc.Config.HostLogin,
 	)
-	if err != nil {
-		tc.ExitStatus = 1
-		return trace.Wrap(err)
+	if connectErr != nil {
+		if tc.shouldGuessCluster(ctx) {
+			// If the user didn't provide a cluster and the auth server says to load all CAs,
+			// try to guess the cluster.
+			nodeClient, err = tc.tryConnectToNodeAnyCluster(ctx, proxyClient, nodeAddrs[0])
+			if err != nil {
+				tc.ExitStatus = 1
+				return trace.NewAggregate(connectErr, err)
+			}
```

Comment:
Sorry I'm not sure I understand - why do we need to guess anything and try connect to any cluster?